syft/execution/plan.py

```diff
@@ -324,12 +387,16 @@ def build(self, *args):
 
         results = (results,) if not isinstance(results, tuple) else results
 
-        arg_ids = [t.id for t in args if isinstance(t, FrameworkTensor)]
         result_ids = [t.id for t in results if isinstance(t, FrameworkTensor)]
 
         for arg in args:
             self.replace_with_placeholders(arg, arg_ids, result_ids, node_type="input")
 
+        for arg in args:
+            # adding the input tag to the tensors, even if nested, adding fcall tag to every argument
+            self.replace_with_placeholders(arg, arg_ids, result_ids, node_type="input")
```

Comment:
This is duplictae with l 393 I think:
`self.replace_with_placeholders(arg, arg_ids, result_ids, node_type="input")`